capstone-app/src/app/modules/maps/map/map.component.ts

```diff
@@ -89,12 +112,47 @@ export class MapComponent implements OnInit{
     this.activeMarkerOutput.emit(this.activeMark);
   }
 
+  getCurrentLocation () {
+    navigator.geolocation.getCurrentPosition(position => {
+      this.position = {
+        lat: position.coords.latitude,
+        lng: position.coords.longitude,
+      };
+      this.location = new google.maps.LatLng(this.position);
+      this.geocoder.geocode({'location': this.position}, (results, status) => {
+        if (status === 'OK') {
+          results.filter(result => {
+            result.types.forEach(type => {
+              if (type == "locality") {
+                this.cityLocation = result.formatted_address;
+              }
+            });
+          });
+        }
+        this.cityOutput.emit(this.cityLocation);
+      });
+      this.placesRequestFunc(this.location);
+    }, () => {
+      this.cityLocation = "Los Angeles, CA, USA"
+        this.location = new google.maps.LatLng(34.0522, -118.2437);
+        this.placesRequestFunc(this.location);
+        this.cityOutput.emit(this.cityLocation);
+    });
```

Comment:
nit/optional: These two lambdas are rather lengthy. What would you think about making these private functions on the class and then passing the functions as arguments here?